src/animation_utils/index.js

```diff
@@ -276,7 +276,8 @@ import software.bernie.geckolib.animation.render.AnimatedModelRenderer;`;
       return result;
   }
 
-  function keyframeExtend(data) {
+  function keyframeExtend(dataIn) {
+    const data = Object.assign({}, dataIn);
```

Comment:
Not strictly related to bugfix but I realized I was mutating the object passed in which seemed like bad form.